packages/core/src/modules/proofs/ProofsApi.ts

```diff
@@ -341,26 +341,23 @@ export class ProofsApi<PPs extends ProofProtocol[]> implements ProofsApi<PPs> {
         autoAcceptProof: options.autoAcceptProof,
         goalCode: options.goalCode,
       })
-
       // Set and save ~service decorator to record (to remember our verkey)
       message.service = ourService
-      await this.didCommMessageRepository.saveOrUpdateAgentMessage(this.agentContext, {
-        agentMessage: message,
-        role: DidCommMessageRole.Sender,
-        associatedRecordId: proofRecord.id,
-      })
-
       await this.messageSender.sendMessageToService(
         new OutboundMessageContext(message, {
           agentContext: this.agentContext,
           serviceParams: {
             service: recipientService.resolvedDidCommService,
             senderKey: ourService.resolvedDidCommService.recipientKeys[0],
-            returnRoute: true,
+            returnRoute: options.useReturnRoute ?? true, // defaults to true if missing
           },
         })
       )
-
+      await this.didCommMessageRepository.saveOrUpdateAgentMessage(this.agentContext, {
+        agentMessage: message,
+        role: DidCommMessageRole.Sender,
+        associatedRecordId: proofRecord.id,
+      })
```

Comment:
Why did you change the order of this? We save the message before sending in all other paths, so this can lead to unexpected behvaiour.

Reply:
The problem is that the `sendMessageToService `method appends the transport decorator to the message. If I save the proof exchange record before sending, that decorator does not get saved which seems wrong to me.

Is there any other way of doing this?

Reply:
I think it's not needed for the outbound message... Otherwise, how did the test pass before this PR?

Reply:
The existence of the decorator is new (something I added) - we never tested for the existence of the decorator before as far as I can see

Reply:
Certainly what I have found is that if I put the ordering back the tests fail (return route undefined, since it can't find the decorator)

Reply:
I think we should remove it for now, as this is the only place where we'd do this, and consistency is more important IMO. In a follow up PR we can change the behaviour, but we should it for all messages.

That said, I think we don't really need to store the decorator. It's nice, but we may refactor the outbound sender quite a bit (and support queues) so we won't know when the message will be actually sent